src/Grids/grid_utils.jl

```diff
@@ -225,121 +226,98 @@ index_range_offset(::Colon, loc, topo, halo)          = - interior_parent_offset
+    ynodes(grid, LX, LY, LZ, with_halos=false)
 
-Keyword argument
-================
-- `reshape`: With `reshape=false` (default) the output is a 1D array while with 
-  `reshape=true` the output is a 3D array with size `1×Ny×1`.
-
-See [`znodes`](@ref) for examples.
+See `znodes` for examples.
 """
-function ynodes(loc, grid; reshape=false)
-
-    y = view(all_y_nodes(loc, grid),
-             interior_indices(loc, topology(grid, 2), grid.Ny))
-
-    return reshape ? Base.reshape(y, 1, length(y), 1) : y
-end
+@inline ynodes(grid, LX::CellLocation, LY::CellLocation, LZ::CellLocation; kwargs...) = ynodes(grid, LY; kwargs...)
 
 """
-    znodes(loc, grid, reshape=false)
-
-Return a view over the interior `loc=Center` or `loc=Face` nodes
-on `grid` in the ``z``-direction. For `Bounded` directions,
-`Face` nodes include the boundary points.
-
-Keyword argument
-================
-- `reshape`: With `reshape=false` (default) the output is a 1D array while with 
-  `reshape=true` the output is a 3D array with size `1×1×Nz`.
-
-Examples
-========
+    znodes(grid, LX, LY, LZ, with_halos=false)
 
 ```jldoctest znodes
 julia> using Oceananigans
 
 julia> horz_periodic_grid = RectilinearGrid(size=(3, 3, 3), extent=(2π, 2π, 1), halo=(1, 1, 1),
                                                  topology=(Periodic, Periodic, Bounded));
 
-julia> zC = znodes(Center, horz_periodic_grid)
+julia> zC = znodes(horz_periodic_grid, Center())
+3-element view(OffsetArray(::StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}, 0:4), 1:3) with eltype Float64:
+ -0.8333333333333334
+ -0.5
+ -0.16666666666666666
+
+julia> zC = znodes(horz_periodic_grid, Center(), Center(), Center())
 3-element view(OffsetArray(::StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}, 0:4), 1:3) with eltype Float64:
  -0.8333333333333334
  -0.5
  -0.16666666666666666
-```
 
-``` jldoctest znodes
-julia> zF = znodes(Face, horz_periodic_grid)
-4-element view(OffsetArray(::StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}, 0:5), 1:4) with eltype Float64:
- -1.0
- -0.6666666666666666
- -0.3333333333333333
-  0.0
+julia> zC = znodes(horz_periodic_grid, Center(), Center(), Center(), with_halos=true)
+-1.1666666666666667:0.3333333333333333:0.16666666666666666 with indices 0:4
 ```
 """
-function znodes(loc, grid; reshape=false)
+@inline znodes(grid, LX::CellLocation, LY::CellLocation, LZ::CellLocation; kwargs...) = znodes(grid, LZ; kwargs...)
 
-    z = view(all_z_nodes(loc, grid),
-             interior_indices(loc, topology(grid, 3), grid.Nz))
 
-    return reshape ? Base.reshape(z, 1, 1, length(z)) : z
+"""
+    reshaped_xnodes(loc, grid; kwargs...)
+"""
+function reshaped_xnodes(grid, loc::CellLocation; kwargs...)
+    x = xnodes(grid, loc; kwargs...)
+    return Base.reshape(x, length(x), 1, 1)
+end
+
+"""
+    reshaped_ynodes(loc, grid; kwargs...)
+"""
```

Comment:
these docstrings are placeholders... we need to fill them up and explain on what these functions actually do, right?

Reply:
@glwagner had comments about this function (thread starts [here](https://github.com/CliMA/Oceananigans.jl/pull/2842#discussion_r1087425320)) and it's not clear to be that it'll remain there. So yes, for now that's a placeholder. Sorry for not making that clear